examples/config.yaml

```diff
@@ -0,0 +1,212 @@
+resources:
+  resource/resourceA:
+    # List of enabled resource detectors. Each detector has a name (FQCN for java), and an optional list of excluded keys. Detector name supports wildcard syntax. Detectors are invoked and merged sequentially. Static attributes and schema_url comprise a resource which is merged last.
+    detectors:
+      - name: com.domain.resources.CustomResourceProvider
+      - name: io.opentelemetry.sdk.extension.resources.*
+        excluded_attribute_keys:
+          # Exclude process.command_line, which is often verbose and may contain secrets
+          - process.command_line
+    # List of static resource attribute key / value pairs.
+    attributes:
+      service.name: my-service
+      service.instance.id: 1234
+    # The resource schema URL.
+    schema_url: http://schema.com
+
+# Limits
+limits:
+  attributes:
+    value_length: 0 # OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT
+    count: 128      # OTEL_ATTRIBUTE_COUNT_LIMIT
+  spans:
+    attributes:
+      value_length: 0 # OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT
+      count: 128      # OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT
+    event:
+      count: 128 # OTEL_SPAN_EVENT_COUNT_LIMIT
+      attributes:
+        count: 128 # OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT
+    link:
+      count: 128 # OTEL_SPAN_LINK_COUNT_LIMIT
+      attributes:
+         count: 128 # OTEL_LINK_ATTRIBUTE_COUNT_LIMIT
+
+# SDK configuration.
+sdk:
+  # Whether the SDK is enabled or not.
+  disabled: false
+  # Configure SDK logging.
+  logging:
+    level: info
+  # Tracer provider configuration.
+  tracer_provider:
+    resource: resource/resourceA # references identifier from resources section
+    # List of span processors, to be added sequentially. Each span processor has a name and args used to configure it.
+    span_processors:
+      # Add simple span processor configured to export with the logging exporter.
+      - name: simple
+        args:
+          # Simple span processor takes exporter as an arg
+          exporter: logging # references identifier from exporters section
+      # Add batch span processor configured to export with the OTLP exporter.
+      - name: batch
+        args:
+          # Batch span processor takes exporter as an arg
+          exporter: otlp/exporterA # references identifier from exporters section
+          # Configure batch span processor batch size and interval settings.
+          max_queue_size: 100
+          scheduled_delay_millis: 1_000
+          export_timeout_millis: 30_000
+          max_export_batch_size: 200
+    # The sampler. Each sampler has a name and args used to configure it.
+    sampler:
+      name: parentbased
+      args:
+        # The parentbased sampler takes root_sampler as an arg, is another sampler itself.
+        root_sampler:
+          name: traceidratio
+          args:
+            ratio: 0.01
+  # Meter provider configuration.
+  meter_provider:
+    resource: resource/resourceA # references identifier from resources section
```

Comment:
Can the resource be different in the TracerProvider/MeterProvider/LoggerProvider?

If not (in Erlang I know it is the same between all three) then I think its an argument for making resource top level and maybe not actually defining "providers" in the config.

Reply:
But it's possible to configure multiple [Tracer|Meter|Logger]Provider for a single application. Will this not be supported by the configuration?

Reply:
I realize that this example does not account for this currently.

Reply:
Ah yea, sorry, this was discussed briefly in slack and @jack-berg agreed we should not support that. But if we do support it then yea, need to have provider sections.

Reply:
I think configuring multiple [Tracer|Meter|Logger]Providers from a single configuration file is not right. If multiple can exist, then they need to be assigned some sort of identifier and be accessible to application code by that identifier. That concept doesn't exist in the SDK and I don't think we should invent it. 

Talked about this in [slack](https://cloud-native.slack.com/archives/C0476L7UJT1/p1667330920134449), but if someone wants multiple providers, they can create multiple configuration files, and have them parsed programatically:
```
OpenTelemetry openTelemetry1 = OpenTelemetryConfigurator.configure("./sdk1.yaml");
OpenTelemetry openTelemetry2 = OpenTelemetryConfigurator.configure("./sdk2.yaml");
```
Then the caller is responsible for wiring the multiple OpenTelemetry instances into their app as they see fit. 

One config file = one configured SDK is a good assumption.

As for having a different resource per provider, I'm inclined to _not_ support that in the file config, and therefore make resource a top level option. OpenTelemetry is a unified approach to traces / metrics / logs - the various telemetry signals are supposed to complement each other (examples include trace context being propagated to exemplars and logs). Resource is supposed to be a bag of identifying and descriptive attributes defining the source of the telemetry. Supporting different resources makes it easier to break the notion of complementary signals, which I think we should discourage.